### PR TITLE
Introduce rspec-puppet-facts for some tests

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
this should get green except for:
* rubocop issues. will be fixed in https://github.com/voxpupuli/puppet-zabbix/pull/217
* 4 spec tests

can somebody review this please? (/me looks at @jyaworski) I've no clue why these for tests fail. Maybe sientifix linux has the wrong osfamily fact.

**Comment by bastelfreak** [comment]
ha awesome... I did the local tests withour STRICT_VARIABLES=yes... is there a nice way to automate the lsb* fact mocking? I would like to get rid of the manual style. Also what is the correct way to deal with the is_pe fact?

File: manifests/repo.pp [hunk 1]
```diff
@@ -93,7 +88,10 @@
           }
         } else {
           $operatingsystem = downcase($::operatingsystem)
-
+          case $::operatingsystemrelease {
+            /\/sid$/ : { $releasename = regsubst($::operatingsystemrelease, '/sid$', '') }
+            default  : { $releasename = $::lsbdistcodename }
```
Comment (jyaworski) [inline manifests/repo.pp]: Is lsbdistcodename a thing on newer facter? I seem to recall the lsb* facts being deprecated.

Comment (bastelfreak) [inline manifests/repo.pp]: Ionly know that it requires the lsb-release package installed and in general the facts are only present on Debian based systems. I was looking for alternative core facts but didn't find any :(

**Pull request merged**